src/redisearch_rs/inverted_index/src/tests.rs

```diff
@@ -485,3 +486,46 @@ fn reading_filter_based_on_field_mask() {
         ]
     );
 }
+
+#[test]
+fn synced_discriminants() {
+    let tests = [
+        (
+            RSResultData::Union(RSAggregateResult::with_capacity(0)),
+            RSResultKind::Union,
+        ),
+        (
+            RSResultData::Intersection(RSAggregateResult::with_capacity(0)),
+            RSResultKind::Intersection,
+        ),
+        (
+            RSResultData::Term(RSTermRecord::default()),
+            RSResultKind::Term,
+        ),
+        (
+            RSResultData::Virtual(RSVirtualResult),
+            RSResultKind::Virtual,
+        ),
+        (
+            RSResultData::Numeric(RSNumericRecord(0.0)),
+            RSResultKind::Numeric,
+        ),
+        (
+            RSResultData::Metric(RSNumericRecord(0.0)),
+            RSResultKind::Metric,
+        ),
+        (
+            RSResultData::HybridMetric(RSAggregateResult::with_capacity(0)),
+            RSResultKind::HybridMetric,
+        ),
+    ];
+
+    for (data, kind) in tests {
+        assert_eq!(data.result_kind(), kind);
+
+        let data_discriminant = unsafe { *<*const _>::from(&data).cast::<u8>() };
```

Comment:
Let's add a safety comment explaining what's going on here.

Reply:
Done in https://github.com/RediSearch/RediSearch/pull/6615/commits/1399c4550c8fbcb22c8a000d834ecddd95c5a370